README.md

```diff
@@ -88,6 +98,7 @@ to skip the dynamic batch sizes in backend.
 * `ENABLE_BATCH_PADDING`: By default an error will be generated if backend receives a request with batch size less than max_batch_size specified in the configuration. This error can be avoided at a cost of performance by specifying `ENABLE_BATCH_PADDING` parameter as `YES`.
 * `RESHAPE_IO_LAYERS`: By setting this parameter as `YES`, the IO layers are reshaped to the dimensions provided in
 model configuration. By default, the dimensions in the model is used.
+* `TARGET_DEVICE`: Choose the OpenVINO device for running the inference. It could be CPU (default), GPU or any of the virtual devices like AUTO, MULTI, HETERO. Note: using Intel GPU is possible only if `--device /dev/dri` is passed to the container and is supported only on linux with x86_64 arch.
```

Comment:
I don't think it is a good idea introducing a model level  TARGET_DEVICE parameter in the model config. Triton allows you to have multiple model instances. And each instance can specify which device to use for the inference. 
So they can have model A, which can specify something like below:
```
  instance_group [
    {
      count: 1
      kind: KIND_GPU
    },
    {
      count: 1
      kind: KIND_CPU
    }
  ]
```

You can rely on information from [TRITONBACKEND_ModelInstanceKind](https://github.com/triton-inference-server/openvino_backend/blob/4016bfde87ec16f993e10c9dc28d0f4fa2357171/src/openvino.cc#L1517) API call to determine the kind of the instance.
If the kind is CPU and the model is not loaded (within model_state_), then load the model on CPU and use it.
If the kind is GPU  and the model is not loaded (within model_state_), then load the model on GPU and use it.
This allows sharing of the model across the model instances.

Reply:
Even if the use-case of loading two separate kinds of model instances is not required, I would still prefer using [TRITONBACKEND_ModelInstanceKind](https://github.com/triton-inference-server/openvino_backend/blob/4016bfde87ec16f993e10c9dc28d0f4fa2357171/src/openvino.cc#L1517) API instance of another config parameter.

Reply:
In openvino, target device contains more options then just the selection between CPU and GPU. It can also set virtual device like MULTI, HETERO or AUTO or BATCH. It can also include extra options like a priority list AUTO:GPU,CPU. My understanding is that settings kind as GPU validated if CUDA GPU is present so that could not be used with other types of devices like iGPU. I couldn't find a way for using KIND as the target device without changes outside of the openvino backend code

Reply:
@ryanloney - proposal in works - will update

Reply:
We are in middle of how to best support non-nvidia hardware with management. Directing this to @nnshah1.

Reply:
I think we can update the model configuration to accommodate these fields. It would help keeping the device specification in a single place. The benefit is that in future we might get other backends targetting these devices besides current openVINO.